vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -1145,8 +1183,14 @@ bool AdsImpl::ShowAd(
       << std::endl << "  url: " << notification_info->url
       << std::endl << "  uuid: " << notification_info->uuid;
 
-  notifications_->Add(*notification_info);
-  ads_client_->ShowNotification(std::move(notification_info));
```

Comment:
Why was this removed?

Reply:
Since it doesn't make sense to store a notification without showing it, I'll put both operations in one function `notifications_->PushBack(*notification_info)` .

Reply:
It is not the responsibility of PushBack to show the notification, please revert the change. By all means add a helper function which does both calls. Otherwise this is not great for adding unit tests.

Reply:
@SergeyZhukovsky A small change so can this please be changed before merging. As we have various devs working with showing notifications right now and this change makes the code harder to understand

Reply:
@tmancey 
It should be one atomic operation: show ad and save it in the queue of posted ads, otherwise it's error prone.  Is there a chance that we save an ad in the queue without showing it?

Reply:
@gdregalo For separation of concerns and the ability to test a unit I disagree as we can add a helper function so both calls happen in one operation

Reply:
@tmancey 
If we talk about concerns and responsibility, IMO, the main functionality of the `class Notifications` is being a wrapper around `std::deque<NotificationInfo> notifications_;` . `Notifications` shouldn't exist at all. 
`notifications_` should be a member of `AdsImpl`. 
Json serialization is a responsibility of a `NotificationInfo`  class. 
At the moment, the functionality is  atomic as it supposed to be. 
How about if I leave it as it is for now and open an issue to refactor it all (including `class Notifications`) to cover your testing concerns later? Otherwise, if i revert it, I will need to test it all again and this story will never end.

Reply:
@gdregalo 👍 I can revert and test as part of follow up PR, there are already tickets raised for decoupling logic from the `AdsImpl` and adding unit tests to cover these areas where we can pick up future improvements